model/ValueObjects/TaskLogStatus.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoTaskQueue\model\ValueObjects;
+
+use Exception;
+use oat\taoTaskQueue\model\TaskLogInterface;
+
+class TaskLogStatus
```

Comment:
Do we really need this? It can a bit misleading regarding task statuses. We have statuses here and in TaskLogInterface. A new developer can be confused which one should be used. I would say, we should used the statuses from TaskLogInterface everywhere and return that to the client and the client can have its logic to aggregate the statuses.

Reply:
I think we should keep it, But I agree with you about TaskLogInterface we have some kind of duplication. 
But this class it's about showing the status to the client, the client only know about 3 states, he does need to know about  if the task it's enqueued or dequeued etc. To leave the client to decide the logic of this it's not ok also, Imagine that we add a new status then we have to contact all the clients and notice about that, in this way from they point of view it's the same. I would rename this to categorize statuses, maybe it would make more sense.